src/logic/cookies/store/reducer/cookies.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+import { Map } from 'immutable'
+import { handleActions, type ActionType } from 'redux-actions'
+import type { Cookie } from '~/logic/cookies/model/cookie'
+import { OPEN_COOKIE_BANNER } from '~/logic/cookies/store/actions/openCookieBanner'
+
+export const COOKIES_REDUCER_ID = 'cookies'
+
+export type State = Map<string, Map<string, Cookie>>
+
+export default handleActions<State, *>(
+  {
+    [OPEN_COOKIE_BANNER]: (state: State, action: ActionType<Function>): State => {
+      const { cookieBannerOpen } = action.payload
+
+      const newState = state.withMutations((map) => {
```

Comment:
there's no real need in `.withMutations` because it performs only a single change of a state. `.withMutations` should be used when you want to perform many changes on the same immutable.js object for performacne reasons:
https://stackoverflow.com/questions/28510753/when-should-i-use-withmutations-on-a-map-in-immutable-js/29381227